### PR TITLE
Toggle between a summarized and expanded view of the error list

### PR DESCRIPTION
**Pull request by hristoramirez**
Adds a summary of errors when the list is collapsed. Allows toggling between the two
Fixes: microsoft/pxt-microbit#2833
summaryoferrorlist

### CODE REVIEW

File: theme/errorList.less [hunk 1]
```diff
@@ -22,7 +22,13 @@
         padding: 1em;
     }
 
-    .noErrorsMessage {
+    .errorListSummary {
         height: unset;
+
+        .summaryMessage:hover {
+            cursor: pointer;
+            color: #00000067;
```
Comment (pelikhan) [inline theme/errorList.less:30]: Use less variables (in future PR)

File: webapp/src/errorList.tsx [hunk 2]
```diff
@@ -42,15 +42,25 @@ export class ErrorList extends React.Component<ErrorListProps, ErrorListState> {
...
+                </div>
+            } else {
+                errorListContent = (errors).map(e =>
+                    <div key={errorKey(e)}>{`${e.messageText} - (${e.line + 1}:${e.column + 1})`}</div>)
+            }
+        } else {
+            errorListContent = <div>{lf("Everything seems fine!")}</div>
+        }
+
+        const errorListInnerClasses = isCollapsed ? "errorListInner errorListSummary" : "errorListInner";
```
Comment (pelikhan) [inline webapp/src/errorList.tsx:59]: We typically do `class1 ${condition ? “Class2” : “”}`

File: webapp/src/errorList.tsx [hunk 3]
```diff
@@ -69,7 +79,8 @@ export class ErrorList extends React.Component<ErrorListProps, ErrorListState> {
 
     onErrorsChanged(errors: pxtc.KsDiagnostic[]) {
         this.setState({
-            errors
+            errors,
+            isCollapsed: errors.length == 0 || this.state.isCollapsed
```
Comment (pelikhan) [inline webapp/src/errorList.tsx:83]: Is errors guaranteed to be defined?

File: webapp/src/errorList.tsx (hunk 2 above)
Comment (hristoramirez) [reply webapp/src/errorList.tsx]: I just changed it by removing the errorListInnerClasses variable and placed it directly, is this what you were meaning?

**Comment by pelikhan** [comment]
Has this PR been merged as part of https://github.com/microsoft/pxt/pull/6921 ?

**Comment by hristoramirez** [comment]
Yeah @pelikhan

**Comment by darzu** [comment]
We can close this PR then. Nice job!

**Pull request closed without merging**